kolibri/core/auth/management/commands/register.py

```diff
@@ -0,0 +1,92 @@
+import sys
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.core.management.base import CommandError
+from morango.certificates import Certificate
+from requests import exceptions
+
+from kolibri.core.auth.utils import confirm_or_exit
+from kolibri.core.auth.utils import get_facility
+from kolibri.core.utils.portal import claim
+
+
+class Command(BaseCommand):
+    help = "Allows registering of facilities which have been synced to a portal server."
+
+    def add_arguments(self, parser):
+        parser.add_argument("token", action="store", type=str, help="Token of project")
+        parser.add_argument(
+            "--facility", action="store", type=str, help="ID of facility to register"
+        )
+        parser.add_argument("--noninteractive", action="store_true")
+
+    def handle(self, *args, **options):
+        facility_id = options["facility"]
+        noninteractive = options["noninteractive"]
+        token = options["token"]
+
+        facility = get_facility(facility_id, noninteractive)
+        # claim the facility
+        try:
+            claim(token, facility)
+            self.stdout.write(
+                "Facility: {} has been successfully registered.".format(facility.name)
+            )
+        except Certificate.DoesNotExist:
+            raise CommandError(
+                "This device does not own a certificate for Facility: {}".format(
+                    facility.name
+                )
+            )
+        # an invalid nonce/claim response
+        except exceptions.HTTPError as e:
+            # handle 404 responses from claim endpoint
+            if e.response.status_code == 404:
+                # if the facility does not exist on data portal, try syncing and retry registering
+                if "Facility" in e.response.text:
```

Comment:
Something about this feels brittle - but at the same time, could the endpoint ever return a 404 for any other reason?

Reply:
maybe JSON-formatted errors with error codes?

Reply:
> maybe JSON-formatted errors with error codes?

That was the suggestion I made at https://github.com/learningequality/kolibri-data-portal/pull/6#discussion_r281357052 (because yes, there are already two different ways it can 404 -- missing Facility, or missing LearningCommunity).